sqlmesh/utils/cache.py

```diff
@@ -99,11 +99,23 @@ def put(self, name: str, entry_id: str, value: T) -> None:
             pickle.dump(value.dict(), fd)
 
     def _cache_entry_path(self, name: str, entry_id: str) -> Path:
+        from sqlmesh.core.state_sync.base import SCHEMA_VERSION
+
         try:
             from sqlmesh._version import __version_tuple__
 
             major, minor = __version_tuple__[0], __version_tuple__[1]
         except ImportError:
             major, minor = 0, 0
-        entry_file_name = f"{name}__{major}__{minor}__{SQLGLOT_MAJOR_VERSION}__{SQLGLOT_MINOR_VERSION}__{entry_id}"
+        entry_file_name = "__".join(
+            [
+                name,
+                str(major),
+                str(minor),
+                SQLGLOT_MAJOR_VERSION,
+                SQLGLOT_MINOR_VERSION,
+                str(SCHEMA_VERSION),
```

Comment:
why don't we have the sqlmesh cache bust on every sqlmesh change, which is more frequent than schema version

Reply:
What do you mean by every sqlmesh change?

Reply:
use sqlmesh version instead of schema version

Reply:
we resolved in chat